### PR TITLE
[ADD][15.0] `purchase_sale_inter_company` without dependency on `stock`

### PR DESCRIPTION
**Pull request by yankinmax**
This module purpose is to split original `purchase_sale_inter_company` **to remove dependency on `stock`**

### CODE REVIEW

File: purchase_sale_inter_company/models/purchase_order.py [hunk 1]
```diff
@@ -103,30 +103,25 @@ def _inter_company_create_sale_order(self, dest_company):
...
+        delivery_address = direct_delivery_address or dest_company.partner_id or False
         new_order = self.env["sale.order"].new(
             {
                 "company_id": dest_company.id,
                 "client_order_ref": name,
-                "partner_id": partner.id,
+                "partner_id": dest_company.partner_id.id,
                 "date_order": self.date_approve,
                 "auto_purchase_order_id": self.id,
             }
```
Comment (yankinmax) [inline purchase_sale_inter_company/models/purchase_order.py:127]: @gurneyalex can you pls take a look, I have doubts about it.
1. `delivery_address = direct_delivery_address or dest_company.partner_id` looks correct to me as we specify destination address on new generated SO
2. `partner_id` on new SO changed to destination company partner. On Invoice creation `Incompatible companies on records` is raised if partner is the same as on the PO. So, for example we got Invoice with company B and partner with company A.

**Comment by yankinmax** [comment]
@gurneyalex @DhavalTforgeFlow 
The error in tests is weird.
```
Draft Invoice INV/2022/00001 (P00009)' belongs to company 'Company B' and 'Partner' (partner_id: 'Company A') belongs to another company
```
We need to generate SO (Company B) from PO (Company A) with `partner_id` who is `partner_id` of (Company A).
Then Invoice is created from generated SO (Company B). Seems legit.
Should we modify `partner_id` of the Invoice to become (Company B) during invoice creation?

**Comment by gurneyalex** [comment]
oh shucks wrong pr

**Review by gurneyalex — APPROVED** [review]

**Review by gurneyalex — CHANGES_REQUESTED** [review]
Can you check the failing test: https://github.com/OCA/multi-company/actions/runs/3363433913/jobs/5576569084#step:8:174

**Comment by yankinmax** [comment]
@gurneyalex I couldn't reproduce the error locally. The test is red since PR is opened.
The error message is: `Draft Invoice INV/2022/00001 (P00009)' belongs to company 'Company B' and 'Partner' (partner_id: 'Company A')`, but the invoice and partner are from the same company in tests. I'll try one more time

**Comment by DhavalTforgeFlow** [comment]
Can you please change the description in README.rst file? It seems like we have old description of configurations when settings were there under Setting --> Companies --> Company. Now it's under general settings (res.config.settings).

**Comment by yankinmax** [comment]
@DhavalTforgeFlow updated with proper README

**Comment by JasminSForgeFlow** [comment]
@yankinmax can you please provide me access to push changes on this PR as well as the purchase_sale_stock_inter_company PR also?

As purchase_sale_inter_company merged in v14 has changes in code and tests for fixing issues related to multi-company in the v14 test.

Thanks

**Comment by yankinmax** [comment]
@JasminSForgeFlow sent invitation, thanks

**Review by mariadforgeflow — APPROVED** [review]
LGTM! Code and functional review:)

**Review by JoanSForgeFlow — APPROVED** [review]

**Comment by yankinmax** [comment]
@JasminSForgeFlow @mariadforgeflow thanks for the work done

**Comment by mariadforgeflow** [comment]
Hi @gurneyalex! Now all tests have passed, could you merge it? Thanks!

**Review by gurneyalex — APPROVED** [review]

**Pull request merged**